sklearn/metrics/pairwise.py

```diff
@@ -1423,6 +1423,13 @@ def pairwise_distances(X, Y=None, metric="euclidean", n_jobs=None, **kwds):
         dtype = bool if metric in PAIRWISE_BOOLEAN_FUNCTIONS else None
         X, Y = check_pairwise_arrays(X, Y, dtype=dtype)
 
+        if metric in PAIRWISE_BOOLEAN_FUNCTIONS \
+                and X.dtype is not bool and Y.dtype is not bool:
+            X, Y = X.astype(np.bool), Y.astype(np.bool)
```

Comment:
Remove this line: the conversion is done in `check_pariwise_array` already

You should use `if dtype == bool and (X.dtype != bool or Y.dtype != bool)` as the condition

Use parenthesis instead of `\`

Reply:
done